tests/docs/test_supported_confs.py

```diff
@@ -0,0 +1,493 @@
+# Copyright 2022 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Test that the supported configurations in the documentation
+matches the supported configurations in the code"""
+import pytest
+import re
+
+import pennylane as qml
+from pennylane import numpy as np
+from pennylane import QuantumFunctionError
+from pennylane.measurements import State, Probability, Expectation, Variance, Sample
+
+pytestmark = pytest.mark.all_interfaces
+
+# import tensorflow as tf
+# import torch
+# import torch.autograd.functional as F
+# import jax
+# from jax import numpy as jnp
+
+tf = pytest.importorskip("tensorflow")
+torch = pytest.importorskip("torch")
+F = pytest.importorskip("torch.autograd.functional")
+jax = pytest.importorskip("jax")
+jnp = pytest.importorskip("jax.numpy")
+
+devices = ["default.qubit"]
+interfaces = [None, "autograd", "jax", "tf", "torch"]
+diff_interfaces = ["autograd", "jax", "tf", "torch"]
+shotss = [None, 100]
+
+diff_methods = ["device", "backprop", "adjoint", "parameter-shift", "finite-diff"]
+return_types = [
+    State,  # scalar cost function of the state
+    "StateVector",  # the state directly
+    "DensityMatrix",
+    Probability,
+    Sample,
+    Expectation,
+    "Hermitian",  # non-standard variant of expectation values
+    "Projector",  # non-standard variant of expectation values
+    Variance,
+]
+
+grad_types = [Expectation, "Hermitian", "Projector", Variance]
+
+
+def get_qnode(interface, diff_method, return_type, shots):
+    """Return a QNode with the given attributes"""
+    dev = qml.device("default.qubit", wires=1, shots=shots)
+
+    @qml.qnode(dev, interface=interface, diff_method=diff_method)
+    def circuit(x):
+        qml.Hadamard(wires=0)
```

Comment:
Indeed, `qml.state()` and `qml.density_matrix()` specifically doesn't support custom wire labels. We would like to change this in the near future.

> adding another configuration axis

I think this would be the way to go. Having said that, the current addition is big enough, so we could also have a separate issue specifically mentioning that the current table could be extended by summarizing the situation for custom wire labels.